src/pykeen/models/inductive/inductive_nodepiece_gnn.py

```diff
@@ -0,0 +1,167 @@
+
+    def __init__(self, edge_index: torch.LongTensor, edge_type: torch.LongTensor) -> None:
+        """Create instance."""
+        super().__init__()
+        self.register_buffer(name="edge_index", tensor=edge_index)
+        self.register_buffer(name="edge_type", tensor=edge_type)
+
+    @classmethod
+    def from_triples(cls, mapped_triples: MappedTriples) -> "BufferedGraph":
+        """Create from mapped triples."""
+        return cls(
+            edge_index=mapped_triples[:, [0, 2]].t(),
+            edge_type=mapped_triples[:, 1],
+        )
+
+
+class InductiveNodePieceGNN(InductiveNodePiece):
+    """Inductive NodePiece with a GNN encoder on top.
+
+    Overall, it's a 3-step procedure:
+
+    1. Featurizing nodes via NodePiece
+    2. Message passing over the active graph using NodePiece features
+    3. Scoring function for a given batch of triples
+
+    As of now, message passing is expected to be over the full graph
+    """
+
+    graphs: Mapping[InductiveMode, BufferedGraph]
+
+    def __init__(
+        self,
+        *,
+        triples_factory: CoreTriplesFactory,
+        inference_factory: CoreTriplesFactory,
+        validation_factory: Optional[CoreTriplesFactory] = None,
+        test_factory: Optional[CoreTriplesFactory] = None,
+        gnn_encoder: Optional[Iterable[nn.Module]] = None,
+        **kwargs,
+    ) -> None:
+        """
+        Initialize the model.
+
+        :param triples_factory:
+            the triples factory. Must have create_inverse_triples set to True.
+        :param gnn_encoder:
+            an interable of message passing layers. Defaults to 2-layer CompGCN with Hadamard composition.
+        :param kwargs:
+            additional keyword-based parameters passed to `InductiveNodePiece.__init__`.
+        """
+        super().__init__(
+            triples_factory=triples_factory,
+            inference_factory=inference_factory,
+            validation_factory=validation_factory,
+            test_factory=test_factory,
+            **kwargs,
+        )
+
+        if gnn_encoder is None:
+            # default composition is DistMult-style
+            dim = self.entity_representations[0].shape[0]
+            gnn_encoder = [
+                CompGCNLayer(
+                    input_dim=dim,
+                    output_dim=dim,
+                    activation=torch.nn.ReLU,
+                    dropout=0.1,
+                )
+                for _ in range(2)
+            ]
+        self.gnn_encoder = nn.ModuleList(gnn_encoder)
+
+        # Saving edge indices for all the supplied splits
+        self.graphs = nn.ModuleDict(
+            {
+                mode: BufferedGraph.from_triples(mapped_triples=factory.mapped_triples)
+                for mode, factory in (
+                    (None, triples_factory),
+                    (TRAINING, inference_factory),
+                    (VALIDATION, inference_factory),
+                    (TESTING, inference_factory),
```

Comment:
It doesn't work for me - the very first line crashes with the error
```
TypeError: module name should be a string. Got NoneType
```

and for the `TRAINING` mode the training `triples_factory` has to be used

Reply:
Do we need `mode=None` anway?

Reply:
ah, the next problem with registering those as modules is overlapping with default pytorch args
```
KeyError: "attribute 'training' already exists"
```

maybe we should return back from this modules idea 😅